pkg/metrics/prometheus.go

```diff
@@ -0,0 +1,152 @@
+package metrics
+
+import (
+	"github.com/grafana/grafana/pkg/log"
+	"github.com/prometheus/client_golang/prometheus"
+	"net/http"
+	"strings"
+)
+
+const (
+	namespace = "grafana"
+	subsystem = "backend"
+)
+
+type PrometheusPublisher struct {
+	CounterCollectors map[string]*prometheus.CounterVec
+	GaugeCollectors   map[string]*prometheus.GaugeVec
+}
+
+func CreatePrometheusPublisher() (*PrometheusPublisher, error) {
+	go func() {
+		http.Handle("/metrics", prometheus.Handler())
+		err := http.ListenAndServe(":9200", nil)
```

Comment:
Oh, forgot to 2 questions

> That works just as well with counters, and I'd expect Prometheus to produce a slightly more accurate answer due to some additional math we do in our rate() function.

That's interesting. Did not think Prometheus could do this accurately (or any TSDB). So if had a counter for user logins recorded every second (scraped or pushed doesnt matter). How would a query in prometheus look like if I wanted to get the total logins per day (or week) for the last year be? can you sum a rate over time? Would not peeks be lost or maybe depends on the step parameter (how Prometheus process queries are still so strange to me)? 

The problem then becomes if you need a TSDB that does roll ups, rolling up a raw counter using last (which is the most common I find) is very in accurate as server restarts will cause roll up to lose a lot of data. The other alternative is to roll up using a derivative / rate function but then the aggregated data becomes a delta and the unaggregated becomes raw which makes it impossible to have the same query target both. 

> Last I checked Graphite was not accurate in this regard, as it can't be. Similarly Prometheus isn't accurate.

This is news to me. How can summing count deltas be inaccurate?